File: client/financial_guide/lib/models/transaction.model.dart
class TransactionModel {
  String userId;
  String type;
  int day;
  String month;
  int year;
  int amount;
  String? category;
  bool? isRecurring = false;
  TransactionModel({required this.userId, required this.type, required this.day, required this.month, required this.year,
      required this.amount, this.category, this.isRecurring});
}

File: server/backend/models/transaction.model.js
const mongoose = require('mongoose');
const UserModel = require("../models/user_model");
const {Schema} = mongoose;

const transactionSchema = new mongoose.Schema({
    userId: {
        type: Schema.Types.ObjectId, 
        ref: UserModel.modelName
    },
    type: {
        type: String, 
        required: true
    },
    day: {
        type: Number, 
        required: true
    },
    month: {
        type: String, 
        required: true,
    },
    year: {
        type: Number, 
        required: true,
    },
    amount: {
        type: Number,
        required: true,
    },
    category: {
        type: String, 
    },
    isRecurring: {
        type: Boolean,
        default: false
    },
});

module.exports = mongoose.model('Transaction', transactionSchema)
